### PR TITLE
feat(quinn-udp): support illumos

### PR DESCRIPTION
**Pull request by AsakuraMizu**
This PR adds support for Illumos and tests on CI. Also fixes a warning on Solaris and Illumos.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -193,8 +211,8 @@ jobs:
   audit:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - uses: EmbarkStudios/cargo-deny-action@v2
+      - uses: actions/checkout@v4
+      - uses: EmbarkStudios/cargo-deny-action@v2
```
Comment (mxinden) [inline .github/workflows/rust.yml:215]: I suggest doing these formatting changes in a separate pull request.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -212,49 +230,49 @@ jobs:
             api-level: 26
 
     steps:
-    - name: Checkout code
```
Comment (mxinden) [inline .github/workflows/rust.yml:215]: Same here, better done in a separate pull request.

**Review by mxinden — COMMENTED** [review]
@psumbera given your expertise with Solaris, want to take a look?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (AsakuraMizu) [reply .github/workflows/rust.yml:215]: I believe it's done accidentally since I have "Format On Save" enabled.

**Comment by psumbera** [comment]
> @psumbera given your expertise with Solaris, want to take a look?

This looks good to me. Thank you!

**Review by Ralith — APPROVED** [review]

File: quinn/src/tests.rs [hunk 3]
```diff
@@ -387,7 +387,10 @@ async fn zero_rtt() {
 }
 
 #[test]
-#[cfg_attr(target_os = "solaris", ignore = "Fails on Solaris")]
+#[cfg_attr(
+    any(target_os = "solaris", target_os = "illumos"),
```
Comment (djc) [inline quinn/src/tests.rs:391]: Nit: could we use `solarish` here, too?

**Review by djc — APPROVED** [review]

File: quinn/src/tests.rs (hunk 3 above)
Comment (mxinden) [reply quinn/src/tests.rs:391]: Friendly ping @AsakuraMizu.

Comment (AsakuraMizu) [reply quinn/src/tests.rs:391]: I suggest doing this in a separate pull request since the `quinn` crate currently does not have a custom build script or cfg aliases setup.
